src/test/java/com/ecommerce/stepDefinitions/CartPageSteps.java

```diff
@@ -0,0 +1,99 @@
+package com.ecommerce.stepDefinitions;
+
+import org.junit.Assert;
+
+import com.ecommerce.base.BaseTest;
+import com.ecommerce.pages.CartPage;
+import com.ecommerce.pages.ProductDetailsPage;
+import com.ecommerce.pages.ProductPage;
+import com.ecommerce.reports.ReportManager;
+
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+
+public class CartPageSteps extends BaseTest {
+	@When("User click on product link")
+	public void user_click_on_product_link() {
+		ProductPage productPage = new ProductPage(driver);
+		productPage.clickOncoriaMateSetProduct();
+		ReportManager.getTest().info("Clicked on product link");
+	}
+
+	@Then("User navigated to the product details page")
+	public void user_navigated_to_the_product_details_page() {
+		ProductDetailsPage productDetailsPage = new ProductDetailsPage(driver);
+		String exceptedProductdetailsPageTitle = "Elegant Korai Floor Mats Set";
+		try {
+			Assert.assertEquals(productDetailsPage.getProductDetailsPageTitle(), exceptedProductdetailsPageTitle);
+			System.out.println("Landed on product details page");
+			ReportManager.getTest().pass("User navigated to the product details page successfully");
+		} catch (AssertionError ae) {
+			System.out.println("Assertion failed" + ae.getMessage());
+			ReportManager.getTest().fail("Failed to land on product details page");
+			throw ae;
+		} catch (Exception e) {
+			System.out.println("An error occured navigating to the product details page " + e.getMessage());
+			ReportManager.getTest().fail("An error occured while navigating to the product details page");
+			throw e;
+		}
+	}
+
+	@When("User click on add to cart button")
+	public void user_click_on_add_to_cart_button() {
+		ProductDetailsPage productDetailsPage = new ProductDetailsPage(driver);
+		productDetailsPage.ClickOnAddProductBtn();
+		ReportManager.getTest().info("Clicked on add to cart button");
+	}
+
+	@Then("User should see item add to cart")
+	public void i_should_see_item_add_to_cart() {
+		ProductDetailsPage procuDetailsPage = new ProductDetailsPage(driver);
+		int ExceptedProductsInShoppingBag = 1;
+		try {
+			Assert.assertEquals(ExceptedProductsInShoppingBag, procuDetailsPage.getProductsInShoppingBag());
+			System.out.println("item added to cart successfully");
+			ReportManager.getTest().pass("item added to cart successfully");
+		} catch (AssertionError ae) {
+			System.out.println("Assertion failed" + ae.getMessage());
+			ReportManager.getTest().fail("Failed to item added to cat");
+			throw ae;
+		} catch (Exception e) {
```

Comment:
use System.out.error instead of System.out.println
do changes in catch block where ever required

Reply:
I updated